schema/crdb/dbinit.sql

```diff
@@ -1572,6 +1552,33 @@ CREATE UNIQUE INDEX IF NOT EXISTS lookup_pool_by_name ON omicron.public.ip_pool
 ) WHERE
     time_deleted IS NULL;
 
+-- The order here is most-specific first, and it matters because we use this
+-- fact to select the most specific default in the case where there is both a
+-- silo default and a fleet default. If we were to add a project type, it should
+-- be added before silo.
+CREATE TYPE IF NOT EXISTS omicron.public.ip_pool_resource_type AS ENUM (
+    'silo'
+);
```

Comment:
I've kept this in even though I took out the possibility of a fleet association to keep the structure general, but note that it does not leak out to the API views — this stuff is on the model, but the API view is called `IpPoolSilo` and just says `silo_id`, no `resource_id` or `resource_type`. I could make that change all the way down here and stop talking about "resources" and "associations". That would clean things up, but it would take more time.